### ISSUE TITLE
Suggestion: Adding Directive

### ISSUE THREAD

**Issue by asyncoder** [aspnet/Blazor#1551]
I'd love to see the ability of supporting also directives, same principle as Angular.

<input MyDirective />

Thanks!

**Comment by RemiBou** [comment]
Honestly, I think one of the serious advantage of Blazor compared to other frameworks is its simplicity : there is a small amount of concept in the framework (page / component). In the angular world you learn a new one every day (https://angular.io/guide/glossary).

When asking a feature maybe we could tell which case cannot be resolved with the current framework state instead of grabbing concept from others. 

Do you have any case in mind where the current framework cannot help you ?

**Comment by asyncoder** [comment]
I wouldn't say grabbing features from others rather than taking advantages. However, the simplicity and reusing C# skills are one of the reasons that Blazor got my heart :-)

Directives are really powerful when you want for example to extend new features and behaviours to an existing components. Let's say that you have a native component "input" and you'd like it behaves somehow when the user focus on it. It wouldn't be so nice to create a new abstract component to do this and then you have to manage all others attributes including accessibility and so on.

Hope I made it clear.

Happy blazoring!

**Comment by srowan** [comment]
Agree, some concept of attribute directives (aka custom bindings in knockout) is critical.    Components are good, but they don't go nearly far enough.

**Comment by asyncoder** [comment]
Any chance to have this feature ASAP? 🙄
@danroth27

**Comment by danroth27** [comment]
@asyncoder This one is a bit further down our backlog right now. We need to first land getting the basics of Razor Components into ASP.NET Core 3.0. But if anyone wants to submit some design proposals for what cross cutting directives in Blazor might look like that would be fine.
